cmd/subnetcmd/join.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package subnetcmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/prompts"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/spf13/cobra"
+)
+
+var avagoConfigPath string
+
+// avalanche subnet deploy
+func newJoinCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "join [subnetName]",
+		Short: "Instruct a validator node to begin validating a new subnet",
+		Long: `Instruct a validator node to begin validating a new subnet.
+The NodeID of that validator node must have been whitelisted by one of the 
+subnet's control keys for this to work.
+
+The node also needs to be restarted.
+If --avalanchego-config is provided, this command tries to edit the config file at that path
+(consider correct permissions).`,
+		RunE: joinCmd,
+		Args: cobra.ExactArgs(1),
+	}
+	cmd.Flags().StringVar(&avagoConfigPath, "avalanchego-config", "", "file path of the avalanchego config file")
+	return cmd
+}
+
+func joinCmd(cmd *cobra.Command, args []string) error {
+	/*
```

Comment:
Because all that is implemented in the addValidator PR (unfortunate PR setup on my end, that part could probably have been an independent PR). As soon as that is merged, I need to uncomment this, and that's when this PR goes from draft to public.

Reply:
Updated to clean code now